### ISSUE TITLE
Demux dies for empty files

### ISSUE THREAD

**Issue by FelixKrueger** [nf-core/demultiplex#142]
### Description of the bug

On a number of occasions we have had the situation that the demux pipeline encounters the case where a sample in the sample sheet doesn't get any reads assigned to it, and the whole process dies. The 'fix' for this is to exclude the offensively empty sample from the samplesheet, and run again. This is a painful exercise, and for single-cell experiments with 384 barcodes per sample it can become unworkable.

Would it be possible to modify the workflow so that empty samples just appear as having a count of 0 reads in a summary table, and downstream processes that fail the entire pipeline are simply skipped?

Many thanks!

**Comment by arnaudceol** [comment]
We have the same issue with @GiampaoloMart, the error is at line 125 of bcl_demultiplexing/main.nf:

```
    **assert line.startsWith('@')**
    line = line.substring(1)
    def fields = line.split(':')
    def rg = [:]
```

The assertion fails because the file is empty. This happens in the Fastp and MultiQC processes.

**Comment by glichtenstein** [comment]
Hi, same issue here, currently I am working in a fix, I still need to update the nf-core tests and pass them, hopefully soon it will succeed and be available in the dev branch (fingers crossed), you can follow at https://github.com/nf-core/modules/pull/4842

**Comment by k1sauce** [comment]
This is fixed by #186

**Comment by k1sauce** [comment]
This is fixed by #186
